packages/shared/src/components/cards/SharePostCard.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { forwardRef, ReactElement, Ref, useRef, useState } from 'react';
+import { Card, CardButton, CardTextContainer, getPostClassNames } from './Card';
+import ActionButtons from './ActionButtons';
+import { SharedPostCardHeader } from './SharedPostCardHeader';
+import { SharedPostText } from './SharedPostText';
+import { SharedPostCardFooter } from './SharedPostCardFooter';
+import { Containter, PostCardProps } from './common';
+
+export const SharePostCard = forwardRef(function SharePostCard(
+  {
+    post,
+    onPostClick,
+    onUpvoteClick,
+    onCommentClick,
+    onBookmarkClick,
+    onMenuClick,
+    onShare,
+    onShareClick,
+    openNewTab,
+    enableMenu,
+    menuOpened,
+    className,
+    children,
+    style,
+    insaneMode,
+    onReadArticleClick,
+    ...props
+  }: PostCardProps,
+  ref: Ref<HTMLElement>,
+): ReactElement {
+  const onPostCardClick = () => onPostClick(post);
+  const [isSharedPostShort, setSharedPostShort] = useState(true);
+  const containerRef = useRef<HTMLDivElement>();
+  const onSharedPostTextHeightChange = (height: number) => {
+    if (!containerRef.current) {
+      return;
+    }
+    setSharedPostShort(containerRef.current.offsetHeight - height < 40);
+  };
+  return (
+    <Card
+      {...props}
+      className={getPostClassNames(post, className, 'min-h-[22.5rem]')}
+      style={style}
+      ref={ref}
+    >
+      <CardButton title={post.title} onClick={onPostCardClick} />
+      <CardTextContainer>
+        <SharedPostCardHeader
+          author={post.author}
+          source={post.source}
+          permalink={post.permalink}
+          onMenuClick={(event) => onMenuClick?.(event, post)}
+          onReadArticleClick={onReadArticleClick}
```

Comment:
Can you please elaborate what this callback does and what arguments are provided to it?
My concern is that we're going to mix some props of the post itself and the shared post object.

Reply:
@sshanzel

Reply:
I think this is a copy-pasted component from the `PostCard` - this is for the screenshot below in our regular feed, I think this part is not necessary here. Not sure though if there is any part where the user should be directed to the actual article. cc @rebelchris 

![image](https://user-images.githubusercontent.com/13744167/215764855-d55a6409-cd56-48a2-8ccf-36c288b649ff.png)

Reply:
I'll look into it and get back on what exactly needs to be done

Reply:
Ok I checked the source code and it's for analytics purposes and reading rank updates. I think it's fine as it is